### PR TITLE
make satisfactoryTypes pure to permit reuse

### PR DESCRIPTION
**Pull request by davidchambers**
This is a step towards addressing #56. I'd like to reuse `satisfactoryTypes` in the proposed `$.test` function, but it would currently require `try`–`catch`, which is clearly unacceptable! ;)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -29,6 +29,30 @@
...
+  var Right = function(x) {
+    return {
+      '@@type': 'sanctuary-def/Either',
+      isLeft: false,
+      isRight: true,
+      chain: function(f) { return f(this.value); },
+      map: function(f) { return Right(f(this.value)); },
+      value: x
+    };
+  };
```
Comment (soroushjp) [inline index.js]: Amazing how simple `Either` implementations are at the end of the day.

**Comment by soroushjp** [comment]
LGTM. Would it make sense to write tests for `_satisfactoryTypes` this function as we change its implementation?

File: index.js (hunk 1 above)
Comment (davidchambers) [inline index.js]: Yep! Simple, principled things make the only reasonable foundation for complex programs. :)

**Comment by svozza** [comment]
Looks good. I'm excited about `$.test`.

**Comment by davidchambers** [comment]
Superseded by #58

**Pull request closed without merging**